File: frontend/src/types/risk.ts
/** 风险相关类型 */

export type RiskLevel = '高' | '中' | '低'

export interface RiskRecord {
  id: string
  contract_no: string
  risk_type: string
  risk_level: RiskLevel
  source_type: string
  detail?: string
  dismissed: boolean
  dismissed_by?: string
  dismissed_at?: string
  created_at: string
}

export interface RiskSummary {
  total: number
  by_level: Record<string, number>
  by_city: Array<Record<string, unknown>>
}

File: backend/app/schemas/risk.py
"""风险相关 schema。"""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class RiskListItem(BaseModel):
    id: UUID
    contract_no: str
    risk_type: str
    risk_level: str
    source_type: str
    detail: str | None = None
    dismissed: bool = False
    dismissed_by: str | None = None
    dismissed_at: datetime | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class DismissRequest(BaseModel):
    dismissed_by: str | None = None


class RiskSummary(BaseModel):
    total: int
    by_level: dict[str, int]
    by_city: list[dict]
